src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java

```diff
@@ -314,28 +317,66 @@ public MethodDefinition build() {
         }
       }
 
-      for (VariableExpr varExpr : method.arguments()) {
-        Preconditions.checkState(
-            varExpr.isDecl(),
-            String.format(
-                "Argument %s must be a variable declaration", varExpr.variable().identifier()));
-      }
-
-      for (TypeNode exceptionType : method.throwsExceptions()) {
-        Preconditions.checkState(
-            TypeNode.isExceptionType(exceptionType),
-            String.format("Type %s is not an exception type", exceptionType.reference()));
-        Preconditions.checkState(
-            !RUNTIME_EXCEPTION_REFERENCE.isAssignableFrom(exceptionType.reference()),
-            String.format(
-                "RuntimeException type %s does not need to be thrown",
-                exceptionType.reference().name()));
-      }
+      performArgumentChecks();
+      performThrownExceptionChecks();
 
       return method;
     }
 
-    void performNullChecks() {
+    private void performArgumentChecks() {
+      // Must be a declaration.
+      arguments().stream()
```

Comment:
This is strange - two subsequent iterations over the same `arguments()` to do validation. Why can't it be done in the same loop?

Reply:
They're done separately to clarify the error messages - separates out `isDecl` and the modifier/scope issues. Alternatively, this could go in a for-loop, but since there are typically very few arguments (e.g. nearly always < 10), the n vs 2n time doesn't really matter in the end.